R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
+    message("Assuming the first unnamed argument to be `rows`.")
+    nams[which(nams == "")[1]] <- "rows"
+    names(args) <- nams
+    n_unnamed_args <- length(which(!nzchar(nams)))
+    all_args_unnamed <- n_unnamed_args == lengt
+  }
+  if (n_unnamed_args == 1 && lengt > 1 && "rows" %in% nams) {
+    message("Assuming the first unnamed argument to be `cols`.")
+    nams[which(nams == "")[1]] <- "cols"
+    names(args) <- nams
+    n_unnamed_args <- length(which(!nzchar(nams)))
+    all_args_unnamed <- n_unnamed_args == lengt
+  }
 
-  if (srow_null) srow <- 0 else srow <- srow - 1L
-  if (scol_null) scol <- 0 else scol <- col2int(scol) - 1L
+  # if 2 unnamed arguments, will be rows, cols.
+  if (n_unnamed_args == 2) {
+    # message("Assuming the first 2 unnamed arguments to be `rows`, `cols` resp.")
+    rows_pos <- which(nams == "")[1]
+    cols_pos <- which(nams == "")[2]
+    nams[c(rows_pos, cols_pos)] <- c("rows", "cols")
+    names(args) <- nams
+    n_unnamed_args <- length(which(!nzchar(nams)))
+    all_args_unnamed <- n_unnamed_args == lengt
+  }
 
-  x <- args[[1]]
-  exp_name <- inherits(x, "data.frame") || inherits(x, "matrix")
+  has_some_unnamed_args <- any(!nzchar(nams))
+  if (has_some_unnamed_args) {
+    stop("Internal error, all arguments should be named after this point.")
+  }
+
+  x_present <- "x" %in% nams
+  cond_acceptable_lengt_1 <- x_present || !is.null(args$start_row) || !is.null(args$start_col)
+
+  if (lengt == 1 && !cond_acceptable_lengt_1) {
+    # Providing a single argument acceptable is only  `x`
+    sentence_unnamed <- ifelse(all_args_unnamed, "unnamed ", " ")
+    stop(
+      "Specifying a single", sentence_unnamed, "argument to `wb_dims()` is not supported.",
+      "\n",
+      "use any of `x`, `start_row` `start_col`. You can also use `rows` and `cols`, You can also use `dims = NULL`"
+    )
+  }
+  cnam_null <- is.null(args$col_names)
+  rnam_null <- is.null(args$row_names)
 
-  # wb_dims(rows, cols)
-  if (length(args) >= 2 && !exp_name) {
-    rows <- 1L
-    cols <- 2L
 
-    # wb_dims(rows = rows, cols = cols)
-    sel <- pmatch(nams, c("rows", "cols"))
-    valid <- length(sel[!is.na(sel)])
-    if (valid == 2) {
-      rows <- sel[rows]
-      cols <- sel[cols]
-    } else if (valid == 1) {
-      stop("found only one cols/rows argument")
+  if (!x_present) {
+    if (!cnam_null || !rnam_null) {
+      stop("`row_names`, and `col_names` should only be used if `x` is present.")
     }
+  }
+  rows_arg <- args$rows
+  #
+  x <- args$x
+  x_has_named_dims <- inherits(x, "data.frame") | inherits(x, "matrix")
+  if (x_has_named_dims && !is.null(rows_arg)) {
+    is_rows_a_colname <- rows_arg %in% colnames(x)
+
+    if (any(is_rows_a_colname)) {
+      stop("`rows` is the incorrect argument. Use `cols` instead.")
+    }
+  }
+  if (is.character(rows_arg)) {
+    warning("It's preferable to specify integers indices for `rows`", "See `col2int(rows)` to find the correct index.")
```

Comment:
This does not look right. `col2int()` is supposed to create integer values from spreadsheet column names. `col2int(cols)` or `as.integer(rows)`?

Reply:
It is just a personal preference warning 

See one of the tests I corrected

It helped catch this.. 

while it makes sense to have cols as a character, It really doesn't make sene to have rows as a character (even if it's supported)